app/models/event.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class Event < ApplicationRecord
+  belongs_to :bag
+  belongs_to :user
+
+  scope :bag, ->(bag_id) { joins(:bag).where("bags.bag_id = :bag_id or external_id = :bag_id", bag_id: bag_id) }
```

Comment:
Might be better implemented as a filter:
`scope :bag, ->(bag_id) { joins(:bag).where(...) if bag_id }`  

This abuses the fact that scopes always return at least the empty/no-op scope.